src/LiveComponent/assets/src/get_array_value.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * Resolve multiple value data from changed HTML element.
+ *
+ * @param element       Current HTML element
+ * @param value         Resolved value of a single HTML element (.value or [data-value])
+ * @param currentValue  Current data value
+ */
+export function getArrayValue(
+    element: HTMLElement,
+    value: string|null,
+    currentValue: any
+): Array<string>|null {
+    // Enforce returned value is an array
+    if (!(currentValue instanceof Array)) {
+        currentValue = [];
+    }
+
+    if (element instanceof HTMLInputElement && element.type === 'checkbox') {
+        const index = currentValue.indexOf(value);
+
+        if (element.checked) {
+            // Add value to an array if it's not in it already
+            if (index === -1) {
+                currentValue.push(value);
+            }
+        } else {
+            // Remove value from an array
+            if (index > -1) {
+                currentValue.splice(index, 1);
+            }
+        }
+    } else if (element instanceof HTMLSelectElement) {
```

Comment:
Should there be a check for multiple attribute?

Reply:
I don't think so - `getArrayValues` is only called in a situation where return value has to be an array (or null), therefore only logic for extracting values as an array is required.

Reply:
Oh, okay, didn't look through refactored `_updateModelFromElement` method :)